accounts/v1/client.go

```diff
@@ -0,0 +1,90 @@
+package accountsv1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+	"golang.org/x/oauth2"
+)
+
+// Client is a wrapper around SAMS primitive REST-based ACcounts API. Most likely, you want
+// to use the more service-to-service "Clients" API instead.
+//
+// This API is needed when using SAMS to identify users, but not perform authorization
+// checks. e.g. the caller will handle its own authorization checks based on the identity
+// of the SAMS user. (The returned User.Subject, the SAMS account external ID.)
+type Client interface {
+	GetUser(ctx context.Context) (*User, error)
+}
+
+// NewClient constructs a new SAMS Accounts client, pointed to the supplied SAMS host.
+// e.g. "https://accounts.sourcegraph.com".
+func NewClient(samsHost string, tokenSource oauth2.TokenSource) Client {
```

Comment:
Alright I don't want to keep stacking things on for you, but in #12 we introduced a mechanism to make it easier to configure SAMS on behalf of clients, for example to leverage internal networking by changing `samsHost` without having to double-check how hosts are configured and avoid accidentally breaking something. Could you look into reconciling the two mechanisms, maybe by moving the conn config to a standalone `sams.ConnConfig` type that can be used across all SAMS clients (`clients/v1`, `accounts/v1`?)

Reply:
I can make a follow up PR too, if you don't mind refactoring your call sites!

Reply:
I would not mind! (But I would like to get this PR merged so I can get started on the SSC-integration to use it.) I'm assuming the env var you will load is `SAMS_URL`? That's the one we use for connecting to the Clients API in the SSC backend.

Reply:
There's several configurations:

https://github.com/sourcegraph/sourcegraph-accounts-sdk-go/blob/b228ef93f415345ed8690f2bc75eab3c8563233c/clientv1.go#L57-L66

Maybe not all of them apply here, but the big ones are `SAMS_URL` and `SAMS_API_URL`